woodwork/accessor_utils.py

```diff
@@ -206,6 +206,11 @@ def wrapper(self, *args, **kwargs):
                 "calling DataFrame.ww.init"
             )
             raise WoodworkNotInitError(msg)
+        diff_cols = set(self._dataframe.columns).difference(
+            set(self._schema.logical_types.keys())
```

Comment:
I think you can do `self._schema.columns.keys()` here which might make it a little more clear that we are comparing the dataframe columns to the columns in the schema.